### ISSUE TITLE
Use changes display units to celsius.

### ISSUE THREAD

**Issue by asp55** [ttimpe/homebridge-midea#11]
Looking at accessory, or sending any commands, in either Homekit or Homebridge ui causes the on unit display to switch to Celsius, regardless of display settings in Homekit / Homebridge ui.

**Comment by ttimpe** [comment]
Yes, this is a known “issue” as the unit is not checked and Celsius is set as default. Should changing the display unit in the Home app change the device’s setting as well? Or do you want it separate from the device’s setting?

**Comment by asp55** [comment]
The setting in HomeKit is labelled as "Hardware Display > Unit" so the implication is that changing that preference in HomeKit would change the display on the hardware.

**Comment by ttimpe** [comment]
After a quick investigation, I do indeed have a way to get the current temperature unit. I do however not yet have way to set the display unit because I don't know which bit to flip. But I'll investigate it.

**Comment by ttimpe** [comment]
Please check if the issue still persists in version 1.1

**Comment by GoxyKG** [comment]
> Yes, this is a known “issue” as the unit is not checked and Celsius is set as default. Should changing the display unit in the Home app change the device’s setting as well? Or do you want it separate from the device’s setting?

For me it use to be Celsius (and I think this issue occured only in last version 1.1.0). 

Changing in Home app is short term, restart of Homebridge service will switch it to Fahrenheit, so it is necessary to manually set to Celsius every time Homebridge service is restarted. 

Additionally, I assume there should be config entry for this, and it can be general, as whoever uses Celsius will use them on all units :)

Thanks in advance.

PS Many thanks for quick development up to very useful version of plugin so far. Respect for your efforts.

**Comment by asp55** [comment]
Checking it on my unit:
Changing the setting in the home app the temperature display toggles to Fahrenheit, but goes back to celsius with the next command sent (such as changing the temperature.)

It's also not preserving the unit setting from the home app. Set "Hardware Display > Unit" to Fahrenheit, check setting again later and it's back to celsius.

**Comment by nihakuto** [comment]
Having a Danby window ac model DAC120B5WDB-6, I have the exact opposite problem. Using some commands on homekit to control the AC set the units back to Fahrenheit (I have it set to Celsius).

For some reason it also resets the Eco mode, but that might be some kind of mode I am not aware of.

**Comment by ttimpe** [comment]
The reason it changes the Eco mode as well is because it's a bit in the same byte (0x14). Which bits sets what mode is unknown to me.
